File: crates/klyron_compat/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron::_compat {

CompatBuilder::CompatBuilder() : config_{} {}

CompatBuilder& CompatBuilder::set_config(const CompatConfig& cfg) {
    config_ = cfg;
    return *this;
}

CompatConfig CompatBuilder::config() const {
    return config_;
}

}

File: crates/klyron_compat/bindings/rust/builder.rs
//! Builder for klyron_compat
use crate::types::CompatConfig;

pub struct CompatBuilder {
    config: CompatConfig,
}

impl CompatBuilder {
    pub fn new() -> Self {
        Self { config: CompatConfig::default() }
    }
    pub fn with_version(mut self, version: &str) -> Self {
        self.config.version = version.to_string();
        self
    }
    pub fn build(self) -> CompatConfig {
        self.config
    }
}
